### PR TITLE
Fix more linter issues

### PR DESCRIPTION
**Pull request by Cxarli**
Only two types of issues remain:

```warning: use of `Debug`-based formatting```
`warning: the function has a cyclomatic complexity of ...`

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -96,14 +98,14 @@ impl ConfigDeserializer {
 
         match deser_result {
             toml::value::Value::Integer(threads) => {
-                if threads > 0 && threads <= MAX_THREADS {
+                if threads > 0 && threads <= i64::from(MAX_THREADS) {
```
Comment (Razican) [inline src/config.rs]: This should probably be changed to `MAX_THREADS as i64`, since it's a lossless casting.

File: src/config.rs [hunk 2]
```diff
@@ -113,19 +115,19 @@ impl ConfigDeserializer {
...
     where
         D: Deserializer<'de>,
     {
         let deser_result: toml::value::Value = serde::Deserialize::deserialize(de)?;
 
         match deser_result {
             toml::value::Value::Table(ref table) => {
-                let criticality_str = table.get("criticality").and_then(|v| v.as_str()).ok_or(
-                    serde::de::Error::custom("Criticality field not found for unknown permission"),
+                let criticality_str = table.get("criticality").and_then(|v| v.as_str()).ok_or_else(
```
Comment (Razican) [inline src/config.rs]: Good catch!

File: src/main.rs [hunk 3]
```diff
@@ -406,7 +404,7 @@ impl<'de> Deserialize<'de> for Criticality {
 
         match deser_result {
             toml::value::Value::String(ref str) => {
```
Comment (Razican) [inline src/main.rs]: Hmm here we should probably not use `str` as a variable name.

File: src/results/mod.rs [hunk 4]
```diff
@@ -138,22 +138,28 @@ impl Results {
...
             }
             Criticality::Critical => {
-                self.critical.insert(vuln);
+                self.critical.insert(vuln)
+            }
+        } {
+            if config.is_verbose() {
+                println!("Failed to insert vulnerability into category, try again or open an issue");
+            } else {
+                println!("A rare issue occured. Please try again");
```
Comment (Razican) [inline src/results/mod.rs]: This message does not really say much. This error should never happen, but if it does, maybe it's a good idea to write some proper error, or even panic (at least in debug mode, with a `debug_assert!`).

File: src/static_analysis/code.rs [hunk 5]
```diff
@@ -62,11 +62,11 @@ pub fn analysis<S: AsRef<str>>(
 
     let handles: Vec<_> = (0..config.get_threads())
         .map(|_| {
-            let thread_manifest = manifest.clone();
-            let thread_files = files.clone();
-            let thread_rules = rules.clone();
-            let thread_vulns = found_vulns.clone();
-            let thread_dist_folder = dist_folder.clone();
+            let thread_manifest = Arc::clone(&manifest);
```
Comment (Razican) [inline src/static_analysis/code.rs:65]: Hmm interesting change, any particular reason?

File: src/static_analysis/manifest.rs [hunk 6]
```diff
@@ -2935,7 +2935,7 @@ impl<'de> Deserialize<'de> for Permission {
 
         match deser_result {
             toml::value::Value::String(ref str) => {
```
Comment (Razican) [inline src/static_analysis/manifest.rs]: Once again, I think the proper thing would be to change the name of the variable. `str` is a standard library type.

**Review by Razican — CHANGES_REQUESTED** [review]
Looks really good! you are working really hard. thanks!

Give a look to my comments below and I will merge as soon as everything is fixed.

File: src/static_analysis/code.rs (hunk 5 above)
Comment (Cxarli) [reply src/static_analysis/code.rs:65]: To be honest, clippy told me that was preferred :laughing:

File: src/results/mod.rs (hunk 4 above)
Comment (Cxarli) [reply src/results/mod.rs]: Saying much more would kinda belong to verbose-mode in my opinion. Maybe something like "An internal error occurred. Please try again"  ?
**Update:** Seems as if this was already fixed in the `develop`-branch in the meantime.

Comment (Razican) [reply src/results/mod.rs]: The thing is that this error should never happen, and if it happens, it depends on the input almost certainly, so trying again won't help. I would say something like `internal error happened, please, run with verbose mode to get more information`, or something along those lines.

**Review by Razican — APPROVED** [review]
I approve the changes, since it loos great, but it seems you forgot some `{` somewhere, as it's not compiling. Could you check?

These are the errors:
```
error: expected `{`, found `}`
   --> target/package/super-analyzer-0.4.1/src/results/mod.rs:176:5
    |
176 |     }
    |     ^

error: unused import: `Criticality`
  --> target/package/super-analyzer-0.4.1/src/results/mod.rs:18:14
   |
18 | use {Config, Criticality, print_warning};
   |              ^^^^^^^^^^^
   |
note: lint level defined here
  --> target/package/super-analyzer-0.4.1/src/main.rs:13:32
   |
13 | #![deny(unused_qualifications, unused, unused_attributes)]
   |                                ^^^^^^
   = note: #[deny(unused_imports)] implied by #[deny(unused)]
```

**Comment by Cxarli** [comment]
~This might seem odd, but it seems to compile and test just fine for me.~
**Update:** Sorry, forgot to pull. Let me try again.
Now I have it too. It seems to come from pulling from develop? ~Let me try building without changes from my side. Okay, it seems to come from my side, let me fix that.~

It seems this happened after manually merging on Github. Should be fixed now.

**Comment by Razican** [comment]
Looks good! the only issue seems with the formatting, could you run `cargo fmt`?

**Comment by Cxarli** [comment]
For some reason Github doesn't seem to react to my commit... Let me try again
**Update:** Really odd... I'll just redo my commits manually I guess

**Comment by Razican** [comment]
I'm closing this for now.

**Pull request closed without merging**